frontend/src/metabase/embedding/embedding-iframe-sdk-setup/utils/build-embed-attributes.ts

```diff
@@ -0,0 +1,166 @@
+import { match } from "ts-pattern";
+
+import {
+  ALLOWED_EMBED_SETTING_KEYS_MAP,
+  ALLOWED_GUEST_EMBED_SETTING_KEYS_MAP,
+  type AllowedEmbedSettingKey,
+} from "metabase/embedding/embedding-iframe-sdk/constants";
+import type {
+  DashboardEmbedOptions,
+  ExplorationEmbedOptions,
+  QuestionEmbedOptions,
+  SdkIframeEmbedBaseSettings,
+} from "metabase/embedding/embedding-iframe-sdk/types/embed";
+import type {
+  SdkIframeDashboardEmbedSettings,
+  SdkIframeEmbedSetupExperience,
+  SdkIframeEmbedSetupSettings,
+  SdkIframeQuestionEmbedSettings,
+} from "metabase/embedding/embedding-iframe-sdk-setup/types";
+import { getVisibleParameters } from "metabase/embedding/embedding-iframe-sdk-setup/utils/get-visible-parameters";
+
+export const buildEmbedAttributes = ({
+  settings,
+  experience,
+  token,
+  wrapWithQuotes,
+}: {
+  settings: SdkIframeEmbedSetupSettings;
+  experience: SdkIframeEmbedSetupExperience;
+  token: string | null;
+  wrapWithQuotes: boolean;
+}) => {
+  const isGuestEmbed = !!settings.isGuestEmbed;
+
+  const settingsWithOverrides = match(experience)
```

Comment:
Is it necessary to list these one-by-one? We already have `ALLOWED_EMBED_SETTING_KEYS_MAP` that lists which keys to white-list. I feel like we could've just do a `pick(settings, ALLOWED_EMBED_SETTING_KEYS_MAP[experience])`, transform each keys to dash-case and be done with it. The only exception is `initialSqlParameters: getVisibleParameters` which we could've easily add.